src/mqtt.cpp

```diff
@@ -489,6 +493,53 @@ mqtt_publish_schedule() {
   }
 }
 
+void
+mqtt_publish_config() {
+  if(!config_mqtt_enabled() || !mqttclient.connected()) {
+    return;
+  }
+  const size_t capacity = JSON_OBJECT_SIZE(128) + 1024;
+  DynamicJsonDocument doc(capacity);
+
+  // Read only information
+  doc["firmware"] = evse.getFirmwareVersion();
+  doc["protocol"] = "-";
+  doc["espflash"] = ESPAL.getFlashChipSize();
+  doc["espinfo"] = ESPAL.getChipInfo();
+  doc["buildenv"] = buildenv;
+  doc["version"] = currentfirmware;
+  doc["evse_serial"] = evse.getSerial();
+  doc["wifi_serial"] = serial;
+
+  // Static supported protocols
+  JsonArray mqtt_supported_protocols = doc.createNestedArray("mqtt_supported_protocols");
+  mqtt_supported_protocols.add("mqtt");
+  mqtt_supported_protocols.add("mqtts");
+
+  JsonArray http_supported_protocols = doc.createNestedArray("http_supported_protocols");
+  http_supported_protocols.add("http");
+
+  // OpenEVSE module config
+  doc["diode_check"] = evse.isDiodeCheckEnabled();
+  doc["gfci_check"] = evse.isGfiTestEnabled();
+  doc["ground_check"] = evse.isGroundCheckEnabled();
+  doc["relay_check"] = evse.isStuckRelayCheckEnabled();
+  doc["vent_check"] = evse.isVentRequiredEnabled();
+  doc["temp_check"] = evse.isTemperatureCheckEnabled();
+  doc["service"] = static_cast<uint8_t>(evse.getServiceLevel());
+  doc["scale"] = evse.getCurrentSensorScale();
+  doc["offset"] = evse.getCurrentSensorOffset();
+  doc["max_current_soft"] = evse.getMaxConfiguredCurrent();
+
```

Comment:
I think all this can be moved to config_serialize (and the equivalant for config_deserialze) this will then be shared between HTTP and MQTT

Reply:
Yes that is exactly what I thought too, this would be a good candidate. 
I hesitated because some comments in the code were indicating that `config_deserialze` is dedicated to the gateway config.  Moving this code inside would enlarge this file to config handling both for gateway and evse then.

I will update!

Reply:
Finsihed. Tested and works fine. I went a little further to also refactor the TODO part that was left ;-)

Reply:
I am now fighting against the build of the divert simulator sub-project 😁

Reply:
... And I won 😀